dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -300,28 +285,20 @@ public String getPageHtml(
                 .setLive(htmlPageUrl.hasLive())
                 .getPageHTML(context.getPageMode());
 
-        if (context.getPageMode() == PageMode.LIVE) {
-            return APILocator.getExperimentsAPI().isAnyExperimentRunning() ?
-                    injectExperimentJSCode(pageHTML, host, request) : injectNoExperimentCode(pageHTML);
+        if (context.getPageMode() == PageMode.LIVE && ConfigExperimentUtil.INSTANCE.isExperimentAutoJsInjection()) {
+            return ExperimentCodeGenerator.INSTANCE.getCode(host, request)
+                    .map(jsCodeToBeInjected -> injectJSCode(pageHTML, jsCodeToBeInjected))
+                    .orElse(pageHTML);
         } else {
             return pageHTML;
         }
     }
 
-    private String injectExperimentJSCode(final String pageHTML, final Host host, final HttpServletRequest request) {
-        return injectJSCode(pageHTML, getExperimentJSCode(host, request));
-    }
-
-    private String injectNoExperimentCode(final String pageHTML) {
-        return injectJSCode(pageHTML, getNotExperimentJSCode());
-    }
-
-
     private String injectJSCode(final String pageHTML, final String JsCode) {
 
         if (StringUtils.containsIgnoreCase(pageHTML, "<head>")) {
-            return StringUtils.replaceIgnoreCase(pageHTML, "<head>",
-                    "<head>" + JsCode);
+            final int indexOf = pageHTML.toLowerCase().indexOf("<head>".toLowerCase());
+            return pageHTML.substring(0, indexOf + 6) + JsCode + pageHTML.substring(indexOf + 6);
```

Comment:
And instead of `indexOf + 6` use:
```
indexOf + HEAD_CONSTANT/length
```

Reply:
done https://github.com/dotCMS/core/pull/24321/commits/3dfbcf6cd202e85cfcdf8cd509c2990d16e05b48#diff-3e34bcc38231d5f5c905331e68d93d63940b6f5e6276f67057d06ca76752651fR305